__tests__/queueGauges.ts

```diff
@@ -1,157 +1,122 @@
-import * as bull from 'bull';
-
+import { Job } from 'bullmq';
 import { getJobCompleteStats, getStats } from '../src/queueGauges';
 
-import { TestData } from './create.util';
+import { makeQueue, makeWorker, TestData } from './create.util';
 import { getCurrentTestHash } from './setup.util';
 
 let testData: TestData;
+const JOB_NAME = 'test-job';
 
 beforeEach(async () => {
-  jest.resetModuleRegistry();
-  const { makeQueue } = await import('./create.util');
-  const hash = getCurrentTestHash();
-  testData = makeQueue(hash);
+	jest.resetModules();
+	const hash = getCurrentTestHash();
+	testData = makeQueue(hash);
+	await testData.queue.waitUntilReady();
 });
 
 afterEach(async () => {
-  await testData.queue.clean(0, 'completed');
-  await testData.queue.clean(0, 'active');
-  await testData.queue.clean(0, 'delayed');
-  await testData.queue.clean(0, 'failed');
-  await testData.queue.empty();
-  await testData.queue.close();
+	await testData.worker?.close();
+	await testData.queue.obliterate({ force: true });
+	await testData.events.close();
+	await testData.queue.close();
 });
 
 it('should list 1 queued job', async () => {
+	const { name, queue, prefix, guages, registry } = testData;
 
-  const {
-    name,
-    queue,
-    prefix,
-    guages,
-    registry,
-  } = testData;
-
-  await queue.add({ a: 1 });
+	await queue.add(JOB_NAME, { a: 1 });
 
-  await getStats(prefix, name, queue, guages);
+	await getStats(prefix, name, queue, guages);
 
-  expect(registry.metrics()).toMatchSnapshot();
+	expect(registry.metrics()).toMatchSnapshot();
 });
 
 it('should list 1 completed job', async () => {
-  const {
-    name,
-    queue,
-    prefix,
-    guages,
-    registry,
-  } = testData;
-
-  queue.process(async (jobInner: bull.Job<unknown>) => {
-    expect(jobInner).toMatchObject({ data: { a: 1 } });
-  });
-  const job = await queue.add({ a: 1 });
-  await job.finished();
-
-  await getStats(prefix, name, queue, guages);
-  await getJobCompleteStats(prefix, name, job, guages);
-
-  expect(registry.metrics()).toMatchSnapshot();
+	const { name, queue, prefix, guages, registry, events } = testData;
+	testData.worker = makeWorker(name, async (jobInner: Job<unknown>) => {
+		expect(jobInner).toMatchObject({ data: { a: 1 } });
+	});
+
+	const job = await queue.add(JOB_NAME, { a: 1 });
+	await job.waitUntilFinished(events);
+
+	await getStats(prefix, name, queue, guages);
+	await getJobCompleteStats(prefix, name, job, guages);
+
+	expect(registry.metrics()).toMatchSnapshot();
 });
 
 it('should list 1 completed job with delay', async () => {
-  const {
-    name,
-    queue,
-    prefix,
-    guages,
-    registry,
-  } = testData;
-
-  queue.process(async (jobInner: bull.Job<unknown>) => {
-    expect(jobInner).toMatchObject({ data: { a: 1 } });
-  });
-  const job = await queue.add({ a: 1 });
-  await job.finished();
-
-  // TODO: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/31567
```

Comment:
@marius-juro was this fixed in bullmq or what's going on here :)?

Reply:
No, from what I could see debugging it's not fixed. Hence they're doing the workaround with `doneJob` 3 lines below. I am going to add the same workaround to the above test `should list 1 completed job`.